### PR TITLE
ELPP-1815 Improved error handling in long running process.

### PR DESCRIPTION
**Pull request by jamiehollern**
@giorgiosironi @nlisgo

### CODE REVIEW

File: src/modules/jcms_notifications/drush/jcms_notifications.drush.inc [hunk 1]
```diff
@@ -95,12 +97,24 @@ function drush_jcms_notifications_article_import($lrp = FALSE) {
...
-        drush_print(dt('Deleted from the queue message !message_id', ['!message_id' => $message->getMessageId()]));
-        $count++;
+        try {
+          $articleVersions = $fetch_service->getArticleVersions($id);
+          $crud_service->crudArticle($articleVersions);
+          drush_print(dt('Processed message !message_id', ['!message_id' => $message->getMessageId()]));
+        }
+        catch (Exception $e) {
+          if ($e instanceof RequestException) {
+            error_log($e);
```
Comment (giorgiosironi) [inline src/modules/jcms_notifications/drush/jcms_notifications.drush.inc]: `error_log` usually accepts strings. Can you extract a few things like `$e->getMessage()`, `$e->getTraceAsString()`

Comment (giorgiosironi) [inline src/modules/jcms_notifications/drush/jcms_notifications.drush.inc]: I would also log all `$e`, not just the `RequestException`.

File: src/modules/jcms_notifications/drush/jcms_notifications.drush.inc [hunk 2]
```diff
@@ -95,12 +97,24 @@ function drush_jcms_notifications_article_import($lrp = FALSE) {
...
+          }
+          else {
+            throw $e;
+          }
+        }
+        finally {
+          $queue_service->deleteMessage($message);
```
Comment (giorgiosironi) [inline src/modules/jcms_notifications/drush/jcms_notifications.drush.inc]: the `finally` ensures that there can be no continuous retry of the same message, nice

File: src/modules/jcms_notifications/drush/jcms_notifications.drush.inc (hunk 1 above)
Comment (giorgiosironi) [inline src/modules/jcms_notifications/drush/jcms_notifications.drush.inc]: or on the contrary:
- `RequestException` with a 404 are unpublished content, can be ignored
- `RequestException` in general are logged
- all other exceptions are logged

**Comment by jamiehollern** [comment]
@giorgiosironi Updated the code. All exceptions will be logged because thanks to our other work there should be no exceptions caused by 404s/500s etc.

**Pull request merged**